conda/core/link.py

```diff
@@ -1429,7 +1446,10 @@ def _calculate_change_report(
                     # noarch: python packages are re-linked on a python version change
                     # just leave them out of the package report
                     continue
-                downgraded_precs[namekey] = (unlink_prec, link_prec)
+                if link_vo == unlink_vo and link_prec.build != unlink_prec.build:
```

Comment:
We should check that build numbers stay the same. The conditions above only check if the build number was greater, but we could a smaller build number here.
```suggestion
                if link_vo == unlink_vo and link.build_number == unlink.build_number and link_prec.build != unlink_prec.build:
```

Reply:
Actually, @jjhelmus brought a good point. Some variants in some packages use build numbers increments (+100, +200, etc) to prioritize features like GPU vs CPU, or types of MPI. In those cases they are not really updates, but adjustments to constraints. 

In that case they should be also "REVISED". Sorry for the back and forth.

Reply:
Sounds good to me, thank @jjhelmus!
Pushed a change to address this :fireworks: